fl4health/clients/fenda_client.py

```diff
@@ -110,8 +124,67 @@ def update_before_train(self, current_server_round: int) -> None:
         assert isinstance(self.model, FendaModel)
         if self.perfcl_loss_weights:
             self.aggregated_global_module = self.clone_and_freeze_model(self.model.global_module)
+        if self.mkmmd_loss_weights and self.old_global_module and self.aggregated_global_module:
+            local_distribution, old_global_distribution, aggregated_distribution = self.update_buffers(
+                local_module=self.model.local_module,
+                global_module=self.old_global_module,
+                aggregated_module=self.aggregated_global_module,
+            )
+            if self.mkmmd_loss_weights[0] != 0.0:
+                self.mkmmd_loss_min.betas = self.mkmmd_loss_min.optimize_betas(
+                    X=old_global_distribution, Y=aggregated_distribution, lambda_m=1e-5
+                )
+            if self.mkmmd_loss_weights[1] != 0.0:
+                self.mkmmd_loss_max.betas = self.mkmmd_loss_max.optimize_betas(
+                    X=local_distribution, Y=aggregated_distribution, lambda_m=1e-5
+                )
+
         return super().update_before_train(current_server_round)
 
+    def update_buffers(
+        self, local_module: torch.nn.Module, global_module: torch.nn.Module, aggregated_module: torch.nn.Module
+    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        """Update the feature buffer of the local, global and aggregated features."""
+        assert isinstance(local_module, torch.nn.Module)
+        assert isinstance(global_module, torch.nn.Module)
+        assert isinstance(aggregated_module, torch.nn.Module)
+
+        local_buffer = []
+        global_buffer = []
+        aggregated_buffer = []
+
+        # Save the state of the local and global modules
```

Comment:
Maybe include a word on why we need not save the `aggregated_modules` training state?